server/src/org/apache/cloudstack/network/topology/BasicNetworkVisitor.java

```diff
@@ -316,4 +317,9 @@ public boolean visit(final StaticRoutesRules staticRoutesRules) throws ResourceU
     public boolean visit(final AdvancedVpnRules vpnRules) throws ResourceUnavailableException {
         throw new CloudRuntimeException("AdvancedVpnRules not implemented in Basic Network Topology.");
     }
+
+    @Override
+    public boolean visit(final QuaggaRules vpnRules) throws ResourceUnavailableException {
+        throw new CloudRuntimeException("AdvancedVpnRules not implemented in Basic Network Topology.");
```

Comment:
CloudRuntimeException is UI friendly and is thrown to pop up the error message on CS UI. It is also consistent with other similar methods. I think if we find strong reasons to change it we should do it for all the other similar methods too.